### PR TITLE
Implemented book, repository, service for book model

### PR DESCRIPTION
**Pull request by dlvsn**

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/mvnw text eol=lf
+*.cmd text eol=crlf
```
Comment (okuzan) [inline .gitattributes:2]: not really relevant for this project, we don't have any .cmd,

File: src/main/java/org/example/onlinebookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Table(name = "books")
+@Data
```
Comment (okuzan) [inline src/main/java/org/example/onlinebookstore/model/Book.java:14]: Don't use Lombok's Data on JPA entities, it contains annotations that can cause problems, use @ Getter @ Setter for now

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job overall

File: src/main/java/org/example/onlinebookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    @Column(nullable = false)
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false, unique = true)
+    private String isbn;
+    @Column(nullable = false)
+    private BigDecimal price;
+    private String description;
+    @Column(name = "cover_image")
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/onlinebookstore/model/Book.java:30]:
```suggestion
```

File: src/main/java/org/example/onlinebookstore/repository/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import org.example.onlinebookstore.exception.DataProcessingException;
+import org.example.onlinebookstore.model.Book;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/onlinebookstore/repository/BookRepositoryImpl.java:13]: use Lombok annotation to generate constructor

File: src/main/java/org/example/onlinebookstore/repository/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+public class BookRepositoryImpl implements BookRepository {
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/onlinebookstore/repository/BookRepositoryImpl.java:20]:
```suggestion
    private final SessionFactory sessionFactory;
```

File: src/main/java/org/example/onlinebookstore/service/BookServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package org.example.onlinebookstore.service;
+
+import java.util.List;
+import org.example.onlinebookstore.model.Book;
+import org.example.onlinebookstore.repository.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/onlinebookstore/service/BookServiceImpl.java:9]: same here

File: src/main/java/org/example/onlinebookstore/service/BookServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+public class BookServiceImpl implements BookService {
+    @Autowired
+    private BookRepository bookRepository;
+
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/onlinebookstore/service/BookServiceImpl.java:16]:
```suggestion
    private final BookRepository bookRepository;
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**